boggs/src/main/java/com/hartwig/healthchecks/boggs/healthcheck/mapping/MappingExtractor.java

```diff
@@ -17,65 +8,73 @@
+                        && p.getFileName().toString().endsWith(REF_SAMPLE_SUFFIX))
+                .findFirst();
+        if (!sampleFile.isPresent()) {
+            throw new FileNotFoundException();
+        }
+        final String externalId = sampleFile.get().getFileName().toString();
+        final Long totalSequences = sumOfTotalSequences(runDirectory);
+        final MappingDataReport mappingDataReport = getFlagstatsData(sampleFile.get(), totalSequences.toString());
+        return new MappingReport(CheckType.MAPPING, externalId, totalSequences.toString(), mappingDataReport);
+    }
 
-	private MappingDataReport getFlagstatsData(@NotNull final Path path, @NotNull final String totalSequences)
-			throws IOException, EmptyFileException {
-		Optional<Path> filePath = Files.walk(new File(path + File.separator + MAPPING + File.separator).toPath())
-				.filter(p -> p.getFileName().toString().endsWith(FLAGSTAT_SUFFIX)
-						&& p.getFileName().toString().contains(REALIGN))
-				.findFirst();
-		if (!filePath.isPresent()) {
-			throw new FileNotFoundException();
-		}
-		FlagStatData flagstatData = flagstatParser.parse(filePath.get().toString());
-		if (flagstatData == null) {
-			throw new EmptyFileException(String.format(EMPTY_FILES_ERROR, path.toString()));
-		}
-		FlagStats passed = flagstatData.qcPassedReads();
+    private MappingDataReport getFlagstatsData(@NotNull final Path path, @NotNull final String totalSequences)
+            throws IOException, EmptyFileException {
+        final Optional<Path> filePath = Files.walk(new File(path + File.separator + MAPPING + File.separator).toPath())
+                .filter(p -> p.getFileName().toString().endsWith(FLAGSTAT_SUFFIX)
+                        && p.getFileName().toString().contains(REALIGN))
+                .findFirst();
+        if (!filePath.isPresent()) {
+            throw new FileNotFoundException();
+        }
+        final FlagStatData flagstatData = flagstatParser.parse(filePath.get().toString());
+        if (flagstatData == null) {
+            throw new EmptyFileException(String.format(EMPTY_FILES_ERROR, path.toString()));
+        }
+        final FlagStats passed = flagstatData.qcPassedReads();
 
-		Double mappedPercentage = toPercentage(passed.mapped() / passed.total());
-		Double properlyPairedPercentage = toPercentage(passed.properlyPaired() / passed.mapped());
-		Double singletonPercentage = passed.singletons();
-		Double mateMappedToDifferentChrPercentage = passed.mateMappedToDifferentChr();
-		Double proportionOfDuplicateRead = toPercentage(passed.duplicates() / passed.total());
-		boolean isAllReadsPresent = passed.total() == ((Double.parseDouble(totalSequences) * 2) + passed.secondary());
-		return new MappingDataReport(mappedPercentage, properlyPairedPercentage, singletonPercentage,
-				mateMappedToDifferentChrPercentage, proportionOfDuplicateRead, isAllReadsPresent);
-	}
+        final Double mappedPercentage = toPercentage(passed.mapped() / passed.total());
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 80 characters (found 87).](https://www.codacy.com/app/hmf/health-checks/file/2780941525/issues/source?bid=3363458&fileBranchId=3365457#l66)